### DISCUSSION TITLE
[✨] Add `trailingIndex` to QwikCity that enables SPA builds to function in Capacitor, Cordova, and other hybrid native app environments

### DISCUSSION THREAD

**Discussion by n8sabes** [QwikDev/qwik-evolution#122] [Proposals For Qwik]
### Is your feature request related to a problem?

To support SPA builds for Capacitor, Cordova and other static hybrid host environments, it is necessary to include `/index.html` to the path. Currently this is not possible without custom link logic sprinkled throughout the entire project.

### Describe the solution you'd like

Enhance QwikCity to include `trailingIndex`.

```ts
export interface QwikCityPlan {
  readonly routes: RouteData[];
  readonly serverPlugins?: RouteModule[];
  readonly basePathname?: string;
  readonly menus?: MenuData[];
  readonly trailingSlash?: boolean;
  readonly trailingIndex?: boolean;    // <-------
  readonly cacheModules?: boolean;
}
```

### Describe alternatives you've considered

Custom `<SmartLink />` components. This works but adds a layer of complexity to every project.

### Additional context

Related to QwikDev/qwik#5731

**Comment by wmertens** [comment]
This makes sense but I think it should be a third state for `trailingSlash`, so `true`, `false`, or `"index"`. If `.html$` is missing from the path, `/index.html` should always be added.

Qwik-city doesn't have any other .html files right? So if you do `/products/[id]` it should result in `/products/chair/index.html`, correct?

**Comment by n8sabes** [comment]
@wmertens, your example path is correct. I agree the third state would be a better solution, but just don't like the variable name but can live with it.

**Comment by devcaeg** [comment]
I am working on a Capacitor integration in Qwik, this enhancement would be good.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
